pandas/tests/reductions/test_reductions.py

```diff
@@ -941,31 +941,45 @@ def test_all_any_params(self):
         with pytest.raises(NotImplementedError, match=msg):
             s.all(bool_only=True)
 
-    def test_all_any_boolean(self):
-        # Check skipna, with boolean type
-        s1 = Series([pd.NA, True], dtype="boolean")
-        s2 = Series([pd.NA, False], dtype="boolean")
-        assert s1.all(skipna=False) is pd.NA  # NA && True => NA
-        assert s1.all(skipna=True)
-        assert s2.any(skipna=False) is pd.NA  # NA || False => NA
-        assert not s2.any(skipna=True)
+    @pytest.mark.parametrize("bool_agg_func", ["any", "all"])
+    @pytest.mark.parametrize("skipna", [True, False])
+    @pytest.mark.parametrize(
+        # expected_data indexed as [[skipna=False/any, skipna=False/all],
+        #                           [skipna=True/any, skipna=True/all]]
+        "data,expected_data",
+        [
+            ([False, False, False], [[False, False], [False, False]]),
+            ([True, True, True], [[True, True], [True, True]]),
+            ([pd.NA, pd.NA, pd.NA], [[pd.NA, pd.NA], [False, True]]),
+            ([False, pd.NA, False], [[pd.NA, False], [False, False]]),
+            ([True, pd.NA, True], [[True, pd.NA], [True, True]]),
+            ([True, pd.NA, False], [[True, False], [True, False]]),
+        ],
+    )
+    def test_any_all_boolean_kleene_logic(
+        self, bool_agg_func, skipna, data, expected_data
+    ):
+        ser = Series(data, dtype="boolean")
+        expected = expected_data[skipna][bool_agg_func == "all"]
 
-        # GH-33253: all True / all False values buggy with skipna=False
-        s3 = Series([True, True], dtype="boolean")
-        s4 = Series([False, False], dtype="boolean")
-        assert s3.all(skipna=False)
-        assert not s4.any(skipna=False)
+        result = getattr(ser, bool_agg_func)(skipna=skipna)
+        assert (result is pd.NA and expected is pd.NA) or result == expected
 
-        # Check level TODO(GH-33449) result should also be boolean
-        s = Series(
+    @pytest.mark.parametrize(
+        "bool_agg_func,expected",
+        [("all", [False, True, False]), ("any", [False, True, True])],
+    )
+    def test_any_all_boolean_level(self, bool_agg_func, expected):
```

Comment:
these are entirely dupilcative of the groupby ones (but i guess ok)

Reply:
Yeah can remove if you'd prefer, this was just to explicitly address #33449